### PR TITLE
config: add support for certificate configuration

### PR DESCRIPTION
**Pull request by codeboten**
Part of https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6351

### CODE REVIEW

**Review by mattsains — COMMENTED** [review]
I'm not sure if you consider HTTP (as opposed to GRPC) config to be in scope for this PR, but the `otlpHTTPLogExporter` has a similar problem - http in this context can also mean https

File: config/log.go
Comment (mattsains) [inline config/log.go:183]: I'm curious about why this was put here and not the outer scope on line 190/191 as a sibling of `if otlp.Config.Compression != nil {` ? the other if statements in the inner block pertain to the result of `url.ParseRequestURI`, so if you think it's better to put otlpConfig checks in this block, it would be good to understand when to add them here and when to add them in the outer scope

**Comment by codeboten** [comment]
> I'm not sure if you consider HTTP (as opposed to GRPC) config to be in scope for this PR, but the otlpHTTPLogExporter has a similar problem - http in this context can also mean https

I was planning on submitting a separate PR for the http exporter

**Comment by mattsains** [comment]
> > I'm not sure if you consider HTTP (as opposed to GRPC) config to be in scope for this PR, but the otlpHTTPLogExporter has a similar problem - http in this context can also mean https
> 
> I was planning on submitting a separate PR for the http exporter

I'll do the rest of the changes if you're okay with that. Some of the other fields require overwriting this change anyway

**Comment by mattsains** [comment]
I created https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6378 which is based on this PR

File: config/log.go
Comment (codeboten) [reply config/log.go]: i ended up moving the code after thinking about this further. originally i thought it made sense to only have it in the case where an endpoint was configured, but after rethinking through it i moved it out at the same level as compression and the other parameters. figured if there was a problem with the cert config it would make sense to surface it regardless of what the endpoint configuration is

**Comment by codeboten** [comment]
Ah I missed your comment, i ended up implementing the http certificate field in this same PR

File: .golangci.yml
Comment (dmathieu) [inline .golangci.yml:74]: Should we move this to be only for the config package, or only for a specific line? Ignoring it globally seems prone to errors.

Comment (codeboten) [reply .golangci.yml:74]: done, is it worth opening an issue to do the same in the core repo?

Comment (dmathieu) [reply .golangci.yml:74]: I would think so, yes.

**Review by dmathieu — APPROVED** [review]

**Comment by codeboten** [comment]
@dmathieu can this be merged? it will unblock https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6378

**Comment by dmathieu** [comment]
We need a second approval to merge.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**
